src/playground/public/assets/js/app.ts

```diff
@@ -107,3 +113,11 @@ function removeFadeInClass() {
   const resultElem = document.getElementById('result')!;
   resultElem.classList.remove('fadeIn');
 }
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function confirmExitFromPlayground(): void {
+  window.addEventListener('beforeunload', function (e) {
+    e.preventDefault();
+    e.returnValue = '';
```

Comment:
I think it would make it easier to maintain if we alway attach this handler in `registerAskScriptEditor` but only execute these lines if `dirty` is `true`

Reply:
add a condition for these handlers